src/ui_interface.h

```diff
@@ -76,6 +76,9 @@ class CClientUIInterface
     /** Show message box. */
     boost::signals2::signal<bool (const std::string& message, const std::string& caption, unsigned int style), boost::signals2::last_value<bool> > ThreadSafeMessageBox;
 
+    /** Ask the user a question, if possible. Falls back to ThreadSafeMessageBox(noninteractive_message, caption, style) if not. */
+    boost::signals2::signal<bool (const std::string& message, const std::string& noninteractive_message, const std::string& caption, unsigned int style), boost::signals2::last_value<bool> > ThreadSafeQuestion;
```

Comment:
Please document the return value in case the question could not be asked of the user.